sklearn/linear_model/_ransac.py

```diff
@@ -538,6 +538,7 @@ def predict(self, X):
             Returns predicted values.
         """
         check_is_fitted(self)
+        self._check_feature_names(X, reset=False)
 
         return self.estimator_.predict(X)
```

Comment:
Why not delegate to the base estimator's own `_validate_data` as we do for `n_features_in_`?

Reply:
We would need to update `fit` to not call `_validate_data`:

https://github.com/scikit-learn/scikit-learn/blob/9580530071de2bb8de3410967884c1a7ce6c3375/sklearn/linear_model/_ransac.py#L277-L283

At a glance I think it is possible.

Reply:
Updated PR with https://github.com/scikit-learn/scikit-learn/pull/18010/commits/a171666d312b3004e4ab4c2f9e72c42d5a1f8f7f that completely delegates validation to the `base_estimator`.

Reply:
Trying to delegate lead to tests failing related to sparse input, because `_safe_indexing` does not support all sparse types:

https://github.com/scikit-learn/scikit-learn/blob/2844f592be6eba36d952a4a1ad68cc41e2845c27/sklearn/linear_model/tests/test_ransac.py#L271-L272

I reverted the change for now

Reply:
Also check_estimator_sparse_data with `_generate_sparse_matrix` generating all the different types of sparse matrices.

Reply:
Thanks for checking.